### PR TITLE
Prepare for Python Modules

### PR DESCRIPTION
**Pull request by ax3l**
Prepare repo structure for more sorted inclusion of python modules. This is still half-way to #1968 but works for now.

Important to know for now:

- `src/tools/`: pre & post-processing for PIConGPU; could become separate repo one day
  - `lib/python/`: path for python *modules* (set `PYTHONPATH` here)
- `examples/SomeName/`
  - `bin/`: example-specific (!) plot scripts that are executable (e.g. using a module and setting ranges)
  - `lib/python/`: example-specific (!) python module (seldom needed)

Also fixes the wrong-named `scripts/` dir in `picongpu` to be named as in install, `bin/`

### CODE REVIEW

**Comment by psychocoderHPC** [comment]
@ax3l Feel free to add a `README.txt` to the new folders with the short description from this PR

**Comment by psychocoderHPC** [comment]
We need to update [this line](https://github.com/ComputationalRadiationPhysics/picongpu/pull/2058/files#diff-baaba04b0773a6bcf8a6d1c04ee53f5fR334) and add the folder `lib`

**Comment by ax3l** [comment]
can do, but that's old `make install` logic when working with `pic-create` which we are not using anymore and should be removed before adding an actual pre-install

File: pic-create [hunk 1]
```diff
@@ -24,7 +24,7 @@ this_dir=`dirname $0`
 #only copy submit and params if we clone from default pic folder
 default_folder_to_copy="submit include/simulation_defines/param include/simulation_defines/unitless"
 #if we clone a project we copy full include 
-folder_to_clone="submit include"
+folder_to_clone="submit include lib/python"
```
Comment (psychocoderHPC) [inline pic-create]: `lib/python` must be `lib` only. Rsync is going recursive throw all folder in lib

Comment (ax3l) [inline pic-create]: uh, and also `bin` for scripts of an example!

File: src/picongpu/submit/hypnos-hzdr/picongpu.profile.example [hunk 2]
```diff
@@ -45,6 +45,8 @@ export PATH=$PATH:$PICSRC
 export PATH=$PATH:$PICSRC/src/splash2txt/build
 export PATH=$PATH:$PICSRC/src/tools/bin
 
+export PYTHONPATH=$PICSRC/src/tools/lib/python:$PYTHONPATH
```
Comment (ax3l) [inline src/picongpu/submit/hypnos-hzdr/picongpu.profile.example]: @ComputationalRadiationPhysics/picongpu-developers please add this line to your `picongpu.profile` if you are working on `dev`

**Pull request merged**